File: repos_estudiantes_ALVAREZ_segunda_entrega/CAGUAYA_APAZA_YONATHAN_TONY/Practica 2/Herencia/ejercicio1/ejercicio1.py
# 1. Modelar diferentes tipos de vehículos. Las clases deben tener las siguientes características:
# Vehículo<marca, modelo, año, precio_base> - Métodos: mostrar_info()
# Coche (hereda de Vehículo)<num_puertas, tipo_combustible> - Métodos: mostrar_info()
# Moto (hereda de Vehículo)<cilindrada, tipo_motor> - Métodos: mostrar_info()
# a) Implementa las clases con sus constructores, getters y setters.
# b) Crea instancias de Coche y Moto y muestra su información usando el método mostrar_info().
# c) Muestra todos los coches que tienen más de 4 puertas.
# d) Mostrar los coches y motos actuales (gestión 2025).

class Vehiculo:
    def __init__(self, marca, modelo, año, precio_base):
        self.__marca = marca
        self.__modelo = modelo
        self.__año = año
        self.__precio_base = precio_base

    def mostrar_info(self):
        return f"{self.__marca} {self.__modelo} - Año: {self.__año} - Precio: {self.__precio_base}"

    def get_marca(self):
        return self.__marca

    def set_marca(self, marca):
        self.__marca = marca

    def get_modelo(self):
        return self.__modelo

    def set_modelo(self, modelo):
        self.__modelo = modelo

    def get_año(self):
        return self.__año

    def set_año(self, año):
        self.__año = año

    def get_precio_base(self):
        return self.__precio_base

    def set_precio_base(self, precio_base):
        self.__precio_base = precio_base

    def inciso_b(self, vehiculos):
        print("Inciso b) Información de todos los vehículos:")
        for v in vehiculos:
            print(v.mostrar_info())

    def inciso_c(self, vehiculos):
        print("Inciso c) Coches con más de 4 puertas:")
        for v in vehiculos:
            if isinstance(v, Coche) and v.get_num_puertas() > 4:
                print(v.mostrar_info())

    def inciso_d(self, vehiculos):
        print("Inciso d) Vehículos del año 2025:")
        for v in vehiculos:
            if v.get_año() == 2025:
                print(v.mostrar_info())


class Coche(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, año, precio_base)
        self.__num_puertas = num_puertas
        self.__tipo_combustible = tipo_combustible

    def mostrar_info(self):
        return f"{super().mostrar_info()} - Puertas: {self.__num_puertas} - Combustible: {self.__tipo_combustible}"

    def get_num_puertas(self):
        return self.__num_puertas

    def set_num_puertas(self, num_puertas):
        self.__num_puertas = num_puertas

    def get_tipo_combustible(self):
        return self.__tipo_combustible

    def set_tipo_combustible(self, tipo_combustible):
        self.__tipo_combustible = tipo_combustible


class Moto(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, año, precio_base)
        self.__cilindrada = cilindrada
        self.__tipo_motor = tipo_motor

    def mostrar_info(self):
        return f"{super().mostrar_info()} - Cilindrada: {self.__cilindrada}cc - Motor: {self.__tipo_motor}"

    def get_cilindrada(self):
        return self.__cilindrada

    def set_cilindrada(self, cilindrada):
        self.__cilindrada = cilindrada

    def get_tipo_motor(self):
        return self.__tipo_motor

    def set_tipo_motor(self, tipo_motor):
        self.__tipo_motor = tipo_motor

vehiculos = [
    Coche("Toyota", "Yaris", 2025, 18000, 4, "Gasolina"),
    Coche("Ford", "Explorer", 2024, 25000, 5, "Híbrido"),
    Moto("Honda", "CBR500R", 2025, 15000, 471, "4T"),
    Moto("Yamaha", "MT-03", 2023, 12000, 321, "4T")
]

vehiculo_aux = Vehiculo("", "", 0, 0)
vehiculo_aux.inciso_b(vehiculos)
vehiculo_aux.inciso_c(vehiculos)
vehiculo_aux.inciso_d(vehiculos)

File: repos_estudiantes_ALVAREZ_segunda_entrega/CAGUAYA_APAZA_YONATHAN_TONY/Practica 2/Herencia/ejercicio1/ejercicio1.java
public class Ejercicio1 {
    public static void main(String[] args) {
        Coche coche = new Coche("Toyota", "Corolla", 2020, 18000, 4);
        Moto moto = new Moto("Yamaha", "FZ", 2022, 9500, 250);

        System.out.println("Información del Coche:");
        coche.mostrarInfo();

        System.out.println("\nInformación de la Moto:");
        moto.mostrarInfo();
    }
}

class Vehiculo {
    private String marca;
    private String modelo;
    private int anio;
    private double precioBase;

    public Vehiculo(String marca, String modelo, int anio, double precioBase) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precioBase = precioBase;
    }

    public void mostrarInfo() {
        System.out.println("Marca: " + marca);
        System.out.println("Modelo: " + modelo);
        System.out.println("Año: " + anio);
        System.out.println("Precio Base: $" + precioBase);
    }

    public String getMarca() { return marca; }
    public void setMarca(String marca) { this.marca = marca; }

    public String getModelo() { return modelo; }
    public void setModelo(String modelo) { this.modelo = modelo; }

    public int getAnio() { return anio; }
    public void setAnio(int anio) { this.anio = anio; }

    public double getPrecioBase() { return precioBase; }
    public void setPrecioBase(double precioBase) { this.precioBase = precioBase; }
}

class Coche extends Vehiculo {
    private int numPuertas;

    public Coche(String marca, String modelo, int anio, double precioBase, int numPuertas) {
        super(marca, modelo, anio, precioBase);
        this.numPuertas = numPuertas;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Número de puertas: " + numPuertas);
    }
}

class Moto extends Vehiculo {
    private int cilindrada;

    public Moto(String marca, String modelo, int anio, double precioBase, int cilindrada) {
        super(marca, modelo, anio, precioBase);
        this.cilindrada = cilindrada;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Cilindrada: " + cilindrada + " cc");
    }
}
